### ISSUE TITLE
QuestieTracker.lua line 791: 'for' limit must be a number

### ISSUE THREAD

**Issue by SpareSimian** [AeroScripts/QuestieDev#1868]
Question 5.10.1. I logged into the IF AH and got this error. Looks like a timer assuming a variable that hasn't yet been initialized. Function is _QuestieTracker:StartFadeTicker() and variable is  _QuestieTracker.highestIndex.

Message: ...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua line 791:
   'for' limit must be a number
Debug:
   [C]: ?
   ...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:791:
      ...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:783
   ..\SharedXML\C_TimerAugment.lua:16:
      ..\SharedXML\C_TimerAugment.lua:14

**Comment by BreakBB** [comment]
Damn I thought I fixed that with the latest release. Thanks for the report I'll have to inspect it closely.

**Comment by BreakBB** [comment]
This is caused by some race condition which is not easy to debug.

**Comment by SpareSimian** [comment]
It seems to be specific to my paladin/smith/miner, and I've only seen it in Ironforge. 

I'd suggest adding a nil check on the for limit in case the background code hasn't initialized it yet.

**Comment by ggreendev** [comment]
I have just seen this. Warlock in Thunder Bluff v5.10.3 including my error information in case it helps narrow things down.
`Date: 2020-05-13 22:24:22
ID: 1
Error occured in: Global
Count: 1
Message: ...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua line 791:
   'for' limit must be a number
Debug:
   [C]: ?
   ...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:791:
      ...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:783
   ..\SharedXML\C_TimerAugment.lua:16:
      ..\SharedXML\C_TimerAugment.lua:14
Locals:

AddOns:
  Swatter, v8.2.6377 (SwimmingSeadragon)
  AucAdvanced, v8.2.6430 (SwimmingSeadragon)
  AucFilterBasic, v8.2.6364 (SwimmingSeadragon)
  AucStatHistogram, v8.2.6366 (SwimmingSeadragon)
  AucStatiLevel, v8.2.6370 (SwimmingSeadragon)
  AucStatPurchased, v8.2.6367 (SwimmingSeadragon)
  AucStatSimple, v8.2.6399 (SwimmingSeadragon)
  AucStatStdDev, v8.2.6369 (SwimmingSeadragon)
  AucUtilFixAH, v8.2.6371 (SwimmingSeadragon)
  AutoBarClassic, v1.13.3.03
  auxaddon, v1.0.0
  Bagnon, v8.3.5
  BeanCounter, v8.2.6434 (SwimmingSeadragon)
  Carbonite, v@project-version@
  CarboniteNotes, v@project-version@
  CarboniteQuests, v@project-version@
  CarboniteSocial, v@project-version@
  CarboniteWarehouse, v@project-version@
  Details, v
  DetailsStreamer, v
  DetailsTinyThreat, v
  Enchantrix, v8.2.6428 (SwimmingSeadragon)
  EnchantrixBarker, v8.2.6469 (SwimmingSeadragon)
  GatherMate2, v1.45.5
  Informant, v8.2.6374 (SwimmingSeadragon)
  Questie, v5.10.3
  SlideBar, v8.2.6375 (SwimmingSeadragon)
  Stubby, v8.2.6376 (SwimmingSeadragon)
  TomCats, v0.0.2
  TradeSkillMaster, vv4.9.35
  TradeSkillMasterAppHelper, vv4.0.8
  BlizRuntimeLib_enUS v1.13.4.11304 <none>
  (ck=48a)

`

**Comment by SpareSimian** [comment]
I suspect the OnEnter handler is being invoked if the mouse pointer happens to be over the tracker display when you create it, but before you're done initializing all the data. So you should set a flag at the _end_ of your construction code to check in the fade-in/fade-out code to see if there's anything ready to be animated.

**Comment by BreakBB** [comment]
@SpareSimian That might be a good hint. I didn't have any time yet to investigate but will start there once I have time.

Thanks everyone for your help and patient towards this.

**Comment by Velinestra** [comment]
Message: ...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:791: 'for' limit must be a number
Time: Thu May 28 19:13:13 2020
Count: 1
Stack: ...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:791: 'for' limit must be a number
...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:791: in function <...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:783>
Interface\SharedXML\C_TimerAugment.lua:16: in function <Interface\SharedXML\C_TimerAugment.lua:14>

Locals: (for index) = 1
(for limit) = nil
(for step) = 1
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "'for' limit must be a number"
_QuestieTracker = <table> {
 ItemButtons = <table> {
 }
 StartFadeTicker = <function> defined @Interface\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:781
 baseFrame = <unnamed> {
 }
 CreateActiveQuestsFrame = <function> defined @Interface\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:416
 expandButton = <unnamed> {
 }
 GetNextItemButton = <function> defined @Interface\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:776
 PrintProgressColor = <function> defined @Interface\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:1040
 GetNextLine = <function> defined @Interface\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:769
 FadeTickerDirection = true
 menuFrame = QuestieTrackerMenuFrame {
 }
 OnDragStop = <function> defined @Interface\AddOns\Questie\Modules\Tracker\QuestieTrackerPrivates.lua:38
 LineFrames = <table> {
 }
 FadeTicker = <table> {
 }
 SetSafePoint = <function> defined @Interface\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:308
 activeQuestsFrame = <unnamed> {
 }
 IsFirstRun = true
 OnDragStart = <function> defined @Interface\AddOns\Questie\Modules\Tracker\QuestieTrackerPrivates.lua:14
 FadeTickerValue = 0.060000
}

**Comment by wallrik** [comment]
Just had this error as well. I was not able to reproduce it after `/reload` or relog.
```
...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:791: 'for' limit must be a number
...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:791: in function <...ce\AddOns\Questie\Modules\Tracker\QuestieTracker.lua:783>
Interface\SharedXML\C_TimerAugment.lua:16: in function <Interface\SharedXML\C_TimerAugment.lua:14>

Locals: (for index) = 1
(for limit) = nil
(for step) = 1
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "'for' limit must be a number"
```

**Comment by BreakBB** [comment]
This will be fixed with the upcoming v6.0.0 release. The tracker got some really good reword seen in #1917

**Comment by BreakBB** [comment]
Release of v6.0.0 is coming very soon. Keep an eye on it
